summarisation_map_reduce.py

```diff
@@ -0,0 +1,56 @@
+from langchain.chains.mapreduce import MapReduceChain
+from langchain.text_splitter import CharacterTextSplitter
+from langchain.chains import ReduceDocumentsChain, MapReduceDocumentsChain
+
+llm = ChatOpenAI(temperature=0)
+
+# Map
+map_template = """The following is a set of documents
+{docs}
+Based on this list of docs, please identify the main themes 
+Helpful Answer:"""
+map_prompt = PromptTemplate.from_template(map_template)
+map_chain = LLMChain(llm=llm, prompt=map_prompt)
+
+# Reduce
+reduce_template = """The following is set of summaries:
+{doc_summaries}
+Take these and distill it into a final, consolidated summary of the main themes. 
+Helpful Answer:"""
+reduce_prompt = PromptTemplate.from_template(reduce_template)
+reduce_chain = LLMChain(llm=llm, prompt=reduce_prompt)
+
+# Takes a list of documents, combines them into a single string, and passes this to an LLMChain
+combine_documents_chain = StuffDocumentsChain(
+    llm_chain=reduce_chain, document_variable_name="doc_summaries"
+)
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `StuffDocumentsChain` class is not imported. This could lead to a `NameError` when trying to create an instance of `StuffDocumentsChain`. Please import the necessary class.
```diff
+ from langchain.chains import StuffDocumentsChain
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->